### ISSUE TITLE
test: apm server managed by EA should not require Fleet anymore

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#10467]
In `8.7`, the sourcemap and agent config logic were reworked to not depend on Fleet anymore. After this, there shouldn't be any technical requirements left why Fleet would be a required component when managed by Elastic Agent. Before we can officially support standalone Elastic Agent, we need to do some manual testing.
Setting up a [standalone Elastic Agent](https://www.elastic.co/guide/en/fleet/8.6/install-standalone-elastic-agent.html) and configure the apm server as integration, and test that
- [x] configuration and confg changes are picked up correctly by the apm-server
- [x] core features are working (data ingestion)
- [x] sourcemap works
- [x] apm agent configuration is working
- [ ] test whether output to logstash is working (not a hard requirement, but good to be aware if there are issues)
- [ ] test whether output to kafka is working (not a hard requirement, but good to be aware if there are issues)

If time permits, also add an automated smoke test for standalone Elastic agent with apm.

**Comment by simitt** [comment]
@kruskall afair from our sync, this is done - could you tick the boxes and close the issue, except if there is anything else left to do.

**Comment by kruskall** [comment]
Ok, I can confirm everything is working and we now have a systemtest for EA managed APM Server.

I haven't tested the kafka/logstash output

**Comment by kruskall** [comment]
> Ok, I can confirm everything is working and we now have a systemtest for EA managed APM Server.

FWIW I thought this was an issue with the agentcfg default credentials but there seems to be some strange issue with the go agent.
To expand on this: I used the go agent and the capture headers option to quickly send requests with curl to a small go program to make sure remote agent config was being applied.

After some investigation, debug logging and changing agent option I confirmed the feature was working but here's the debug log from the APM Go agent:

```
{"level":"debug","message":"central config update: updated capture_body to all"}
{"level":"warning","message":"central config failure: unsupported config: capture_headers"}
{"level":"debug","message":"central config update: updated log_level to debug"}
```

The capture headers doesn't seem to be recognized so the settings in the APM tab has no effect. I don't think this is an issue with the Elastic agent managed APM Server but I don't think this is intended.
